src/plugins/query_enhancements/server/routes/index.ts

```diff
@@ -85,8 +85,14 @@ export function defineSearchStrategyRouteProvider(logger: Logger, router: IRoute
           const queryRes: IDataFrameResponse = await searchStrategy.search(context, req as any, {});
           return res.ok({ body: { ...queryRes } });
         } catch (err) {
+          let error;
+          try {
+            error = JSON.parse(err.message);
```

Comment:
the status code might be in err object, not in err.message. If err.message is a valid string, then status code is lost